pkg/api/mock/mock_api_implementation.go

```diff
@@ -231,8 +232,12 @@ func (a *mockAPI) Create(ctx context.Context, o types.Object, opts ...types.Crea
 
 // Update overwrites a types.Object in MockAPIs local storage
 func (a *mockAPI) Update(ctx context.Context, o types.IdentifiedObject, opts ...types.UpdateOption) error {
-	a.mu.Lock()
-	defer a.mu.Unlock()
+	for _, h := range a.hooks[preUpdateHook] {
+		h(ctx, a, o)
+	}
+
+	a.dataMu.Lock()
```

Comment:
Shouldn't the lock happen before the execution of the hooks?

Reply:
I've fixed it after our Mattermost conversation, thanks for that!